### PR TITLE
simplify the dialing logic in the h2quic client, fix mutex in roundtripper held to long

### LINKED ISSUE #620: mutex held to long in h2quic.QuicRoundTripper
in `QuicRoundTripper.getClient` the mutex is locked while a connection is dialed.

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #620.

### CODE REVIEW

File: h2quic/client.go [hunk 1]
```diff
@@ -137,8 +132,8 @@ func (c *client) handleHeaderStream() {
 	c.mutex.Unlock()
 }
 
-// Do executes a request and returns a response
-func (c *client) Do(req *http.Request) (*http.Response, error) {
+// Roundtrip executes a request and returns a response
+func (c *client) RoundTrip(req *http.Request) (*http.Response, error) {
 	// TODO: add port to address, if it doesn't have one
```
Comment (lucas-clemente) [inline h2quic/client.go]: Is this TODO still relevant? If so, do we have an issue?

File: h2quic/client.go [hunk 2]
```diff
@@ -147,14 +142,16 @@ func (c *client) Do(req *http.Request) (*http.Response, error) {
 		return nil, fmt.Errorf("h2quic Client BUG: RoundTrip called for the wrong client (expected %s, got %s)", c.hostname, req.Host)
 	}
 
-	hasBody := (req.Body != nil)
+	c.dialOnce.Do(func() {
+		c.handshakeErr = c.dial()
```
Comment (lucas-clemente) [inline h2quic/client.go]: Can we make handshakeErr a local variable here?

Comment (marten-seemann) [inline h2quic/client.go]: No, because the Do is only executed once, but the error has to be returned from all calls to Roundtrip.

Comment (lucas-clemente) [inline h2quic/client.go]: Right – Go doesn't have static vars. Sorry :)

**Pull request merged**